### PR TITLE
fix(Array) Array in property access

### PR DESCRIPTION
**Pull request by mohammedzamakhan**
#333

### CODE REVIEW

**Comment by mohammedzamakhan** [comment]
@Krustie101

File: demo/hello.ts [hunk 1]
```diff
@@ -360,12 +360,20 @@ export class HelloApp {
...
-        key: 'nested.arrays.0',
+        key: 'nested.arrays.0.item',
+        type: 'input',
+        templateOptions: {
+          label: 'Array with dots',
+        },
+        defaultValue: 'Default Value A',
+        parsers: [(value) => (value || '').toUpperCase()],
+      }, {
+        key: 'nested.arrays[1].item',
```
Comment (aitboudad) [inline demo/hello.ts]: I'm not in favor of this, isn't `nested.arrays.1.item` enough ?
also this make us consistent with angular form

Comment (aitboudad) [inline demo/hello.ts]: the easy way to solve this without lot of changes is replacing `[` and `]` by `.`
`nested.arrays[1].item` -> `nested.arrays.1.item`

Comment (mohammedzamakhan) [inline demo/hello.ts]: I just wanted to keep it consistant with angular-formly, also most developers are used to use [] for arrays

Comment (aitboudad) [inline demo/hello.ts]: make sense :+1:

**Pull request merged**